sdk/identity/azure-identity/src/main/java/com/azure/identity/DefaultAzureCredentialBuilder.java

```diff
@@ -271,25 +270,47 @@ private void loadFallbackValuesFromEnvironment() {
     }
 
     private ArrayList<TokenCredential> getCredentialsChain() {
+        Configuration configuration = identityClientOptions.getConfiguration() == null
+            ? Configuration.getGlobalConfiguration().clone()
+            : identityClientOptions.getConfiguration();
+        String selectedCredentials = configuration.get("AZURE_TOKEN_CREDENTIALS");
+        boolean useProductionCredentials = true;
+        boolean useDeveloperCredentials = true;
+        if (!CoreUtils.isNullOrEmpty(selectedCredentials)) {
+            if ("prod".equalsIgnoreCase(selectedCredentials)) {
+                useDeveloperCredentials = false;
+            } else if ("dev".equalsIgnoreCase(selectedCredentials)) {
+                useProductionCredentials = false;
```

Comment:
nit: curious why we default both modes to true and disable the other one if a mode value is found. I'd have expected the other way around.

Reply:
It made sense to me since they are both default to on. I flipped it around; I'm not sure I like it better. What do you think?

Reply:
I'm fine either way, I'm just thinking as we expand it, starting from true may get harder to manage.